src/Instrumentation/Psr18/src/Psr18Instrumentation.php

```diff
@@ -0,0 +1,112 @@
+                if (!$request instanceof RequestInterface) {
+                    Context::storage()->attach(Context::getCurrent());
+
+                    return null;
+                }
+
+                static $instrumentation;
+                $instrumentation ??= new CachedInstrumentation('io.opentelemetry.contrib.php.psr18', schemaUrl: TraceAttributes::SCHEMA_URL);
+                $propagator = Instrumentation\Globals::propagator();
+                $parentContext = Context::getCurrent();
+
+                $spanBuilder = $instrumentation
+                    ->tracer()
+                    ->spanBuilder(sprintf('HTTP %s', $request->getMethod()))
+                    ->setParent($parentContext)
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::HTTP_URL, (string) $request->getUri())
+                    ->setAttribute(TraceAttributes::HTTP_METHOD, $request->getMethod())
+                    ->setAttribute(TraceAttributes::HTTP_FLAVOR, $request->getProtocolVersion())
+                    ->setAttribute(TraceAttributes::HTTP_USER_AGENT, $request->getHeaderLine('User-Agent'))
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_CONTENT_LENGTH, $request->getHeaderLine('Content-Length'))
+                    ->setAttribute(TraceAttributes::NET_PEER_NAME, $request->getUri()->getHost())
+                    ->setAttribute(TraceAttributes::NET_PEER_PORT, $request->getUri()->getPort())
+                    ->setAttribute(TraceAttributes::CODE_FUNCTION, $function)
+                    ->setAttribute(TraceAttributes::CODE_NAMESPACE, $class)
+                    ->setAttribute(TraceAttributes::CODE_FILEPATH, $filename)
+                    ->setAttribute(TraceAttributes::CODE_LINENO, $lineno)
+                ;
+
+                foreach ($propagator->fields() as $field) {
+                    $request = $request->withoutHeader($field);
+                }
+                foreach ((array) (get_cfg_var('otel.instrumentation.http.request_headers') ?: []) as $header) {
+                    if ($request->hasHeader($header)) {
+                        $spanBuilder->setAttribute(sprintf('http.request.header.%s', strtr(strtolower($header), ['-' => '_'])), $request->getHeader($header));
+                    }
+                }
+
+                $span = $spanBuilder->startSpan();
+                $context = $span->storeInContext($parentContext);
+                $propagator->inject($request, HeadersPropagator::instance(), $context);
+
+                Context::storage()->attach($context);
+
+                return [$request];
+            },
+            post: static function (ClientInterface $client, array $params, ?ResponseInterface $response, ?Throwable $exception): void {
+                $scope = Context::storage()->scope();
+                $scope?->detach();
+
+                if (!$scope || $scope->context() === Context::getCurrent()) {
```

Comment:
As I mentioned during review of PSR-15, it's not clear for me why there is part after or operator `$scope->context() === Context::getCurrent()` ?